### PR TITLE
make the migration table name configurable

### LINKED ISSUE #1125: Make migration table name "schema_migrations" configurable
Follow up from https://github.com/elixir-lang/ecto/issues/566
Something like `config :app, App.Repo, migration_table: "my_migrations"` ?

### PR DESCRIPTION
**Pull request by tony612**
resolves #1125

### CODE REVIEW

File: test/support/test_repo.exs [hunk 1]
```diff
@@ -120,3 +120,9 @@ defmodule Ecto.TestRepo do
 end
 
 Ecto.TestRepo.start_link(url: "ecto://user:pass@local/hello")
+
+Application.put_env(:ecto, Ecto.TestSchemaRepo, [migration_table: "my_schema_migrations"])
+
+defmodule Ecto.TestSchemaRepo do
```
Comment (josevalim) [inline test/support/test_repo.exs]: We can define this repository directly in the test case that uses it. :)

**Comment by josevalim** [comment]
Thank you ! I have added one tiny comment!

**Comment by tony612** [comment]
@josevalim Updated :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by jamesvl** [comment]
Small tip: If using this feature with the ecto 2.0 beta, the config parameter was renamed from `migration_table` (as seen in this commit) to `migration_source` (per the ecto 2.0.0 changelog).

(adding comment in case searches continue to send people to this PR when looking at this topic)

**Pull request merged**